src/sagemaker/jumpstart/utils.py

```diff
@@ -813,24 +833,90 @@ def get_jumpstart_model_id_version_from_resource_arn(
     return model_id, model_version
 
 
-def extract_info_from_hub_content_arn(
+def get_info_from_hub_resource_arn(
     arn: str,
-) -> Tuple[Optional[str], Optional[str], Optional[str], Optional[str]]:
-    """Extracts hub_name, content_name, and content_version from a HubContentArn"""
+) -> HubArnExtractedInfo:
+    """Extracts descriptive information from a Hub or HubContent Arn."""
 
-    match = re.match(constants.HUB_MODEL_ARN_REGEX, arn)
+    match = re.match(constants.HUB_CONTENT_ARN_REGEX, arn)
     if match:
-        hub_name = match.group(4)
+        partition = match.group(1)
         hub_region = match.group(2)
-        content_name = match.group(5)
-        content_version = match.group(6)
-
-        return hub_name, hub_region, content_name, content_version
+        account_id = match.group(3)
+        hub_name = match.group(4)
+        hub_content_type = match.group(5)
+        hub_content_name = match.group(6)
+        hub_content_version = match.group(7)
+
+        return HubArnExtractedInfo(
+            partition=partition,
+            region=hub_region,
+            account_id=account_id,
+            hub_name=hub_name,
+            hub_content_type=hub_content_type,
+            hub_content_name=hub_content_name,
+            hub_content_version=hub_content_version,
+        )
 
     match = re.match(constants.HUB_ARN_REGEX, arn)
     if match:
-        hub_name = match.group(4)
+        partition = match.group(1)
         hub_region = match.group(2)
-        return hub_name, hub_region, None, None
+        account_id = match.group(3)
+        hub_name = match.group(4)
+        return HubArnExtractedInfo(
+            partition=partition,
+            region=hub_region,
+            account_id=account_id,
+            hub_name=hub_name,
+        )
+
+    return None
+
+
+def construct_hub_arn_from_name(
```

Comment:
consider moving this function to https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/jumpstart/session_utils.py

Reply:
this utils module is a bit more general than just for hubs. Also, the module's already close to 1K lines long. So i'd put these functions in `session_utils` or maybe make a new file `hub_utils`